msteams-platform/toolkit/use-CICD-template.md

```diff
@@ -10,233 +10,374 @@ ms.date: 04/20/2022
-3. Create a new Microsoft Entra application secret.
+1. Run the pipeline.
 
-> [!TIP]
-> Save your tenant id, application id (AZURE_SERVICE_PRINCIPAL_NAME), and the secret (AZURE_SERVICE_PRINCIPAL_PASSWORD) for future use.
+    Push code to the repo to trigger pipeline.
 
-For more information, see [Azure service principals guidelines](/azure/active-directory/develop/howto-create-service-principal-portal). The following are the three ways to create service principals:
+    > [!NOTE]
+    > There's no need to commit env files under env/ folder to the repo. The env variables required for executing the CI/CD pipeline are already established in the pipeline variables.
 
-* [Microsoft Azure portal](/azure/active-directory/develop/howto-create-service-principal-portal)
-* [Windows PowerShell](/azure/active-directory/develop/howto-authenticate-service-principal-powershell)
-* [Microsoft Azure CLI](/cli/azure/create-an-azure-service-principal-azure-cli)
+    After the pipeline executes successfully, the log displays that the code is deployed to Azure and the appPackage is generated in the artifacts.
 
-## Publish Teams app using Teams Developer Portal
+    :::image type="content" source="../assets/images/teams-toolkit-v2/published.png" alt-text="Screenshot shows the pipeline runs successfully.":::
 
-If there are any changes related to Teams app's manifest file, you can update the manifest and publish the Teams app again. To publish Teams app manually, you may use [Developer Portal for Teams](https://dev.teams.microsoft.com/home).
+### Set up CI/CD pipelines using your own workflow
 
-Perform the following steps to publish your app:
+If the Teams App CLI doesn't meet your pipeline requirements, you can develop a custom deployment process that suits your needs. This section provides guidance on deploying to Azure with custom methods. If you're using a different cloud platform.
 
-1. Sign in to [Developer portal for Teams](https://dev.teams.microsoft.com) using the corresponding account.
-2. Import your app package in zip, select **App** > **Import app** > **Replace**.
-3. Select the target app in app list.
-4. To publish your app, select **Publish** > **Publish to your org**.
+> [!NOTE]
+> If you already have a complete CI/CD pipeline for deploying to your Azure resource, and your Teams app needs to read environment variables during runtime, configure these environment variables in the settings of your Azure resource. For testing post-deployment, see [generate the appPackage for the teams app](#generate-the-apppackage-for-the-teams-app).
+
+The `teamsapp deploy` command executes the actions defined in the `deploy` stage of the `teamsapp.yml` file. The `deploy` stage consists of `build` and `deploy` actions. To create a custom deployment method, rewrite these actions based on your specific requirements and preferences.
+
+A basic bot TypeScript project as an example, the deploy stage in its `teamsapp.yml` is as follows:
```

Comment:
```suggestion
Taking a basic bot TypeScript project as an example, the deploy stage in its `teamsapp.yml` is as follows:
```

Reply:
Updated as "As an example, a basic bot TypeScript project has the following deploy stage in its `teamsapp.yml`:"